app/src/main/java/com/github/studydistractor/sdp/ui/DistractionStatScreen.kt

```diff
@@ -18,12 +18,22 @@ import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.testTag
-import com.github.studydistractor.sdp.procrastinationActivity.DistractionStatInterface
+import com.github.studydistractor.sdp.distraction.DistractionViewModel
+import com.github.studydistractor.sdp.procrastinationActivity.FirebaseDistractionStat
+import com.google.firebase.auth.FirebaseAuth
 
 @Composable
-fun DistractionStatScreen(uid : String, did : String, stat : DistractionStatInterface) {
+fun DistractionStatScreen(viewModel : DistractionViewModel) {
     val feedback = remember {mutableStateOf("")}
     val tag = remember {mutableStateOf("")}
+    //TODO CHANGE TO DID OF THE DISTRACTION
+    val stat = FirebaseDistractionStat(viewModel.distraction!!.name!!)
+    //TODO CHANGE TO DID OF THE DISTRACTION
+    val did = viewModel.distraction!!.name!!
+    var uid = ""
+    if (FirebaseAuth.getInstance().uid != null){
+        uid = FirebaseAuth.getInstance().uid!!
+    }
     Column(
```

Comment:
It would be nice if that looked a bit better, e.g. the elements are centered etc. It's also not clear what the text fields are for. When i press "+", the app crashes with an Illegal Argument Exception.

Reply:
Yes I agree with you. I will try to do it in a better way.